app/controllers/dashboard_controller.rb

```diff
@@ -659,16 +659,19 @@ def start_url_for_user(start_url)
   end
 
   def session_reset
-    # Clear session hash just to be sure nothing is left (but copy over some fields)
-    winh    = session[:winH]
-    winw    = session[:winW]
-    referer = session['referer']
+    # save some fields to recover back into session hash after session is cleared
+    keys_to_restore = [:winH, :winW, :referer, :browser, :user_TZO]
+    data_to_restore = {}
+
+    keys_to_restore.each do |k|
+      data_to_restore[k] = session[k]
+    end
 
     session.clear
 
-    session[:winH]     = winh
-    session[:winW]     = winw
-    session['referer'] = referer
+    keys_to_restore.each do |k|
+      session[k] = data_to_restore[k]
+    end
```

Comment:
why not the simpler `session = data_to_restore` or `session = data_to_restore.dup`?